collection-scripts/gather_monitoring

```diff
@@ -1,29 +1,235 @@
+GAHTER_MONIT_QUERY_STEP_DEFAULT="1m"
+GATHER_MONIT_DISCOVERY_GRAFANA_DEFAULT="yes"
+MONIT_NS="openshift-monitoring"
+MONIT_ENV="must-gather-env"
+
 BASE_COLLECTION_PATH="../must-gather"
 MONITORING_PATH="${BASE_COLLECTION_PATH}/monitoring"
+PROM_PATH="${MONITORING_PATH}/prometheus"
+GF_PATH="${MONITORING_PATH}/grafana"
+test -d ${PROM_PATH} || mkdir -p ${PROM_PATH}
+test -d ${GF_PATH} || mkdir -p ${GF_PATH}
+
+PROM_HOST=$(oc -n ${MONIT_NS} get route prometheus-k8s -o jsonpath='{.spec.host}{"\n"}')
+PROM_URL="https://${PROM_HOST}"
+
+GF_HOST=$(oc -n ${MONIT_NS} get route grafana  -o jsonpath='{.spec.host}{"\n"}')
+GF_URL="https://${GF_HOST}"
+
+OCP_TOKEN=$(oc whoami -t)
+
+JQ_PATH=$(which jq 2>/dev/null)
+CURL_OPTIONS="-sk"
+
+echo "INFO: Get custom metrics from ConfigMap ${MONIT_ENV} on project ${MONIT_NS}"
+oc -n ${MONIT_NS} get cm ${MONIT_ENV} -o jsonpath='{.data.env}' > ${MONITORING_PATH}/env 2> /dev/null
+if [[ $? -eq 0 ]]; then
+    echo "INFO: Loading custom environments variables from ${MONITORING_PATH}/env"
+    source ${MONITORING_PATH}/env
+else
+    echo "INFO: Unable to load custom environments from ConfigMap, ignoring."
+fi
+
+_START=${GAHTER_MONIT_START_DATE:-${GAHTER_MONIT_START_DATE_DEFAULT}}
+DATE_START=$(date -d "${_START}" +%s)
+DATE_END=$(date +%s)
+# If the resolution is too low, the API will deny due to 11k limitation
+QUERY_STEP=${GAHTER_MONIT_QUERY_STEP:-${GAHTER_MONIT_QUERY_STEP_DEFAULT}}
+
+echo "INFO: Metrics config: from=${DATE_START} to=${DATE_END} step=${QUERY_STEP}"
+echo "INFO: Config Env GAHTER_MONIT_START_DATE: ${GAHTER_MONIT_START_DATE}"
+echo "INFO: Config Env GAHTER_MONIT_QUERY_STEP: ${GAHTER_MONIT_QUERY_STEP}"
+echo "INFO: Config Env GATHER_MONIT_CUSTOM_METRICS: ${GATHER_MONIT_CUSTOM_METRICS}"
+echo "INFO: Config Env GATHER_MONIT_DISCOVERY_PREFIXES: ${GATHER_MONIT_DISCOVERY_PREFIXES}"
+echo "INFO: Config Env GATHER_MONIT_DISCOVERY_GRAFANA: ${GATHER_MONIT_DISCOVERY_GRAFANA}"
+echo "INFO: Prometheus endpoint URL: ${PROM_URL}"
+echo "INFO: Grafana  endpoint URL: ${GF_URL}"
+
+function install_jq() {
+
+    if [[ -f ${JQ_PATH} ]]; then
+        return
+    fi
+
+    # install jq from binary (required to Dashnoard parser)
+    export JQ_PATH=/usr/local/bin/jq
+    curl -s -o ${JQ_PATH} \
+      http://stedolan.github.io/jq/download/linux64/jq && \
+      chmod +x ${JQ_PATH}
+}
+
+# Make a query (single metric) to Prometheus API /api/v1/query_range
+function prom_query_range() {
+    QUERY=$1
+    METRIC_FILE=${2:-$(echo ${QUERY} |awk -F'\(' '{print$1}')}
+
+    # The metrics are ziped to decrease the datatransfer
+    curl ${CURL_OPTIONS} \
+        -H "Authorization: Bearer ${OCP_TOKEN}" \
+        -H 'Accept-encoding: gzip' \
+        --data-urlencode "query=${QUERY}" \
+        --data-urlencode "start=${DATE_START}" \
+        --data-urlencode "end=${DATE_END}" \
+        --data-urlencode "step=${QUERY_STEP}" \
+        $PROM_URL/api/v1/query_range > ${PROM_PATH}/metric-${METRIC_FILE}.json.gz
```

Comment:
Nit: Double quote $PROM_URL and ${METRIC_FILE}. Also add curly braces for PROM_URL for consistency.